cmd/dep/status.go

```diff
@@ -304,18 +348,24 @@ type MissingStatus struct {
 	MissingPackages []string
 }
 
-func runStatusAll(ctx *dep.Ctx, out outputter, p *dep.Project, sm gps.SourceManager) (bool, bool, error) {
+// errorStatus contains information about error and number of status failures.
+type errorStatus struct {
+	err   error
+	count int
+}
+
+func runStatusAll(ctx *dep.Ctx, out outputter, p *dep.Project, sm gps.SourceManager) (bool, bool, errorStatus) {
```

Comment:
I wonder if the `errorStatus` type is worth it.  The return lines are sort of complex. Did you consider adding a return value instead? `(bool, bool, int, error)`

In either case, this function could probably use some documentation regarding what is returned. Actually, a case could be made for naming them. They are all declared in highly visible scope already anyways (bools wouldn't change), plus returns would be simpler.
Maybe `(digestMismatch bool, hasMissingPkgs bool, errCnt int, err error)` or something?

Reply:
I thought of returning the error count, but then thought it would look weird. Returning error count when there's no error. If you think it's fine, we can return the count 😊

Yes, the return needs documentation. I'll add those variable names for the existing variables, will add error count later if we agree that we need it.

Reply:
> I thought of returning the error count, but then thought it would look weird. Returning error count when there's no error. If you think it's fine, we can return the count :blush:

I don't think there is a problem with this in principle, as long as the documentation makes the purpose clear.  The default assumption tends to be that a non-nil err means any other return values are irrelevant, but `io.Writer`, for example, always returns the bytes written.